artemis-server/src/main/java/org/apache/activemq/artemis/core/persistence/impl/journal/JournalStorageManager.java

```diff
@@ -464,7 +464,9 @@ private void sendJournalFile(JournalFile[] journalFiles, JournalContent type) th
       for (JournalFile jf : journalFiles) {
          if (!started)
             return;
```

Comment:
Replicator currentReplicator = replicator;
if (currentReplicator != null) ...


There's no lock in place ... you could have a case where it satisfies the if and you still get the NPE.. I have seen it happening before in similar cases.